frontend/src/views/DashboardPage/components/MoveSecrets/DirectorySelector.tsx

```diff
@@ -0,0 +1,143 @@
+                        if (!parentItem.folders.find(folder => folder.id === currentItem.id)) {
+                            parentItem.folders.push(currentItem);
+                        }
+                    }
+
+                } else if (currentItem.id !== queryFolderId) {
+                    $directoryTree.push(currentItem);
+                }
+
+                i += 1;
+            }
+            setDirectoryTree(() => $directoryTree)
+        }
+    }, [directoryData])
+
+    const handlePathSelection = (path: string, folderId: string) => {
+        setSelectedFolder(folderId)
+        setSelectedPath(path);
+        onSelect(path, folderId);
+    };
+
+
+    const renderRootDirectory = (folders: Folder[] | undefined, currentPath: string, depth = 0) => {
+        if (!folders) return null;
+
+        const indentation = depth * 20;
+
+        return (
+            <ul style={{ marginLeft: `${indentation}px` }}>
+                {folders.map((folder) => (
+                    <li key={Math.floor(Math.random() * 50000) + 1}>
+                        <button
+                            className={twMerge(`py-1 px-2 bg-mineshaft-500  hover:bg-mineshaft-600 mb-2 rounded-md ${selectedFolder === folder.id ? "bg-mineshaft-400 hover:bg-mineshaft-400" : ""}`)}
+                            onClick={() => handlePathSelection(`${currentPath}/${folder.name}`, folder.id)}
+                            type="submit"
+                        >
+                            {folder.name}
+                        </button>
+                        {folder.folders && renderRootDirectory(folder.folders, `${currentPath}/${folder.name}`, depth + 1)}
+                    </li>
+                ))}
+            </ul>
+        );
+    };
+
+    const renderCurrentDirectory = (folders: Folder[] | undefined, currentPath: string) => {
+        if (!folders) return null;
+
+        return (
+            <ul className="mt-3">
+                {folders.map((folder) => (
+                    <li key={folder.id}>
+                        <button
+                            className="py-1 px-2 bg-mineshaft-500 mb-2 hover:bg-mineshaft-600 text-gray-400 rounded-md"
+                            onClick={() => handlePathSelection(`${currentPath}/${folder.name}`, folder.id)}
+                            type="submit"
+                        >
+                            {folder.name}
+                        </button>
+                    </li>
+                ))}
+            </ul>
+        );
+    };
+
+    const adjustedPath = selectedPath.replace(/\/root/g, "/").replace(/\/{2,}/g, "/");
+
+
+    return (
+        <div className="p-4 border border-mineshaft-600 rounded-lg shadow-md">
+            {
+                (directoryTree.length > 0) && (
+                    <div>
+                        <h2 className="text-md font-semibold  text-gray-300">Select a folder path within the root directory:</h2>
+                        {renderRootDirectory(directoryTree, "")}
```

Comment:
I dont see the purpose of these functions other than too much rendering from functions. Can you change this pattern to just placing the html here

Reply:
For this, I am thinking of using useCallback for the function to memoise it and avoid unneccessary re-renders.. I think I prefer the functions being outside. It can make the code harder to read if I just embed the HTML there. what do you think?

Reply:
Hey @EBEN4REAL .Thats an unncessary optimization. `useCallback` also comes at a cost. Why not just keep the html here itself if its not being reused and just put a comment

Reply:
I have removed for  it for rendering the folders in current directory. However, this function for the root folder will still be needed as it is a recursive function.